domain_scraper/scrape_domains.py

```diff
@@ -0,0 +1,221 @@
+"""Argument parser for domain-scraper module"""
+
+import email
+import json
+import logging
+import os
+import shutil
+import ssl
+import smtplib
+import re
+import sys
+from argparse import Namespace
+from collections import defaultdict
+from datetime import date
+from email.message import Message
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from time import time
+
+import pkg_resources
+from jinja2 import Template
+
+
+from .config import DEVELOPMENT as CONFIG
+from .parser import parse_arguments
+
+logger = logging.getLogger(__name__)
+
+
+class MessageDomains:
+    """Represents scraped domains from regular email Message object"""
+
+    REGEX_FROM = re.compile(r"from\s+(.+?)\s+(by|via|with|id|for|;)")
+    REGEX_BY = re.compile(r"by\s+(.+?)\s+(from|via|with|id|for|;)")
+
+    def __init__(self, msg: Message):
+        self.msgid = msg["Message-ID"].strip("\t\n<>")
+        received_headers = msg.get_all("Received")
+        domains_set = set()  # use set to avoid duplicates
+        for received_header in received_headers:
+            if domain := self.REGEX_FROM.search(received_header):
+                domains_set.add(domain.group(1))
+            if domain := self.REGEX_BY.search(received_header):
+                domains_set.add(domain.group(1))
+        self.domains = list(domains_set)
+
+    def __str__(self) -> str:
+        domains_str = "\n\t".join(self.domains)
+        return f"MsgID {self.msgid}:\n\t{domains_str}"
+
+
+def make_sure_dirname_exists(file: str) -> None:
```

Comment:
make_sure_dirname_exists -> is_dir_exists

Reply:
`is_dir_exist` sounds a bit unclear to me. Maybe `mkdir_for_file` or `make_parent_dir`?